subprojects/chrome-trace/src/main/java/org/gradle/trace/GradleTracingPlugin.java

```diff
@@ -25,18 +25,36 @@
     private final GCMonitoring gcMonitoring = new GCMonitoring();
     private final BuildOperationListenerAdapter buildOperationListener;
 
-    private GradleTracingPlugin(GradleInternal gradle, File traceFile) {
+    private GradleTracingPlugin(GradleInternal gradle, File traceFolder, String traceFilePattern) {
         this.buildRequestMetaData = gradle.getServices().get(BuildRequestMetaData.class);
-        traceResult = new TraceResult(traceFile);
+        traceResult = new TraceResult(getTraceFile(traceFolder, traceFilePattern));
         systemMonitoring.start(traceResult);
         gcMonitoring.start(traceResult);
         buildOperationListener = BuildOperationListenerAdapter.create(gradle, traceResult);
         gradle.addListener(new TraceFinalizerAdapter(gradle));
     }
 
+    private File getTraceFile(File traceFolder, String traceFilePatter) {
+        String phase = System.getProperty("org.gradle.profiler.phase.display.name");
+        String build = System.getProperty("org.gradle.profiler.number");
+        int invocation = 1;
+        File traceFile;
+        do {
+            String traceFileName = traceFilePatter.replace("{phase}", phase)
```

Comment:
I think it would make sense that the `start` method only takes the file base name, and then would add this postfix. This way we can avoid passing in a pattern altogether. WDYT?

Reply:
Good point, will fix it

Reply:
Fixed